### PR TITLE
Added ability to resume download. Added support for Java 1.8.

### PR DESCRIPTION
**Pull request by justinlittman**

### CODE REVIEW

**Comment by justinlittman** [comment]
Steps to test:
1. Replace `settings/config.properties` with `config.properties` from shared_configs. Set `outputBaseDir` to point to scratch space.
2. `./gradlew installDist`
3. `./build/install/wasapi-downloader/bin/wasapi-downloader --collectionId 5425 --crawlStartAfter 2018-01-01`
4. Download some files and then interrupt. Notice which files successfully downloaded and which is partially downloaded.
5. ./build/install/wasapi-downloader/bin/wasapi-downloader --collectionId 5425 --crawlStartAfter 2018-01-01 --resume`

Expected result:
* Partially downloaded file is deleted and downloading is repeated.
* Already existing files are skipped. For example:
```
Cli-dl-7328-02d5:wasapi-downloader jlittman$ ./build/install/wasapi-downloader/bin/wasapi-downloader --collectionId 5425 --crawlStartAfter 2018-01-01 --resume
file already retrieved: https://warcs.archive-it.org/webdatafile/ARCHIVEIT-5425-TEST-JOB555436-20180309195356729-00000.warc.gz
file already retrieved: https://warcs.archive-it.org/webdatafile/ARCHIVEIT-5425-QUARTERLY-JOB694543-SEED1050159-20181027205452505-00001-h3.warc.gz
```

File: build.gradle [hunk 1]
```diff
@@ -35,15 +35,25 @@ dependencies {
...
+    compile 'com.sun.xml.bind:jaxb-core:2.3.0'
+    compile 'com.sun.xml.bind:jaxb-impl:2.3.0'
+
     //Unit testing framework.
     testCompile 'junit:junit:4.12'
 
     //For creating mock objects in unit tests
-    testCompile "org.mockito:mockito-core:2.+"
+    // See https://github.com/powermock/powermock/wiki/Mockito#supported-versions
+//    testCompile "org.mockito:mockito-core:2.8.0-2.8.9"
```
Comment (jcoyne) [inline build.gradle]: Cruft?

File: build.gradle [hunk 2]
```diff
@@ -35,15 +35,25 @@ dependencies {
...
+    testCompile 'org.mockito:mockito-core:2.23.0+'
 
     // need PowerMock for mocking constructors (local objects within methods)
-    testCompile "org.powermock:powermock-module-junit4:1.7+"
-    testCompile "org.powermock:powermock-api-mockito2:1.7+"
+//    testCompile "org.powermock:powermock-module-junit4:1.7+"
```
Comment (jcoyne) [inline build.gradle]: Cruft?

**Review by jcoyne — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (justinlittman) [reply build.gradle]: Decrufted.

**Pull request merged**